### DISCUSSION TITLE
How to resolve 'Cannot find module' and 'noEmit' issues in TypeScript project with Vite?

### DISCUSSION THREAD

**Discussion by MuhammadSaim** [shadcn-ui/ui#4151] [Q&A]
Cannot find module: I'm encountering an error Cannot find module `@/lib/utils` even though I have configured path aliases in both `tsconfig.json` and `vite.config.ts`. It seems like TypeScript or Vite isn't resolving the alias properly.

![image](https://github.com/shadcn-ui/ui/assets/19898499/a17842a7-3fac-4782-983a-cadb90b0b162)

<kbd>tsconfig.json</kbd>
```json
{
  "files": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.node.json"
    }
  ],
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

<kbd>vite.config.ts</kbd>
```ts
import react from "@vitejs/plugin-react";
import path from "path";
import { defineConfig } from "vite";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      "@": path.resolve(__dirname, "./src"),
    },
  },
});
```

I still couldn't get the alias right If I import using relative directory path working fine but the @ alias has an error.

**Comment by Loureto** [comment]
@MuhammadSaim

Could you do the following: move the configuration into the "tsconfig.app.json" file and add the following inside "compilerOptions":

`"baseUrl": ".",
"paths": {
  "@/*": ["./src/*"]
}`

**Reply by MuhammadSaim** [reply]
Thanks for the answer It fixes my issue now, I have another one when I generate the component, shadcn generates in the root path `@/components/ui` instead of `src/components/ui`. 

![image](https://github.com/shadcn-ui/ui/assets/19898499/cf84d31d-f2e5-488c-bac1-e6e9d42fbf86)

**Comment by Loureto** [comment]
@MuhammadSaim 

In my project, I copied everything from the "tsconfig.app.json" and "tsconfig.node.json" files and added it all to the main file, which is "tsconfig.json".

From my point of view, it doesn't make much sense to create two configuration files, as it will depend on the size of each project.

It turned out like this:

![image](https://github.com/shadcn-ui/ui/assets/34563016/edea6ac9-f46e-43a2-85e0-c5bbf0629cdc)

![image](https://github.com/shadcn-ui/ui/assets/34563016/563f1e23-43c7-4365-8050-53d038a50ff0)

![image](https://github.com/shadcn-ui/ui/assets/34563016/0ef222d7-1f16-403b-a0f5-17ef25f6cff8)

![image](https://github.com/shadcn-ui/ui/assets/34563016/b94965f8-beb7-462d-813c-47747d00df1a)

![image](https://github.com/shadcn-ui/ui/assets/34563016/435098a1-5938-414b-8a0a-4588f3f41b40)

**Comment by inwardmovement** [comment]
Hi, I have the same error, even after applying @Loureto's fix.
Can you take a look at the _alias_ branch of [this repo](https://github.com/inwardmovement/bookmarks-palette/tree/alias) and tell me what's wrong please?
